### ISSUE TITLE
[Feature Request] LUKS encryption of destination drive with LVM.

### ISSUE THREAD

**Issue by MarioLawrence** [stevenshiau/clonezilla#90]
First of all, great job. Thank you for all you do. 

Moving on to my request...I maintain and handle the deployment of a set of Linux images for my users through Puppet. 
Our security baselines require that I encrypt the drive of deployed laptops. 

Previously, I would manually load a base Ubuntu desktop onto the device, install puppet agent, input necessary criteria, and then allow Puppet to configure the system. This process took a while. And for laptops it takes a little longer. I would use Ubuntu's Ubiquity installer to encrypt the drive on initial install, and then do the aforementioned steps afterwards.

I've shaved down deployment time considerably by using Clonezilla. 
I created a VM to create a test image. As generic as possible. I disabled Puppet, removed the FQDN, and left realm. A perfect NAKED image of our managed Ubuntu. 
Clonezilla creates the image, which I have saved to a secure and dedicated NAS via smb. 

Now, all I need to do is use Clonezilla on my target device, aim at the image on the smb, and install to the drive. I've reduced deployment time by 72%. A fantastic result enabled by Clonezilla! I'm most pleased. 

The problem I'm encountering is that I need total drive encryption for laptops. And while Clonezilla is capable of creating an encrypted IMAGE, my security baselines do NOT allow me to deploy an encrypted image on to multiple laptops. Because they would all have the same LUKS Master Key. This is a huge NO-NO for the InfoSec guys. 

My request is that Clonezilla perform encryption on the target drive when it's deploying an image. This way each encrypted deployment has its own unique LUKS Master Key. LVM (logical volume management) will simply help maximize security and will enable me to meet the security baselines of InfoSec. 

TL;DR - 
Please provide LUKS encryption and LVM support for deploying images to to destination drives. Clonezilla currently ONLY allows encryption when CREATING images, but not when deploying them. 

If Clonezilla already has the functionality and I'm just not aware of it, please point me in the right direction! Thanks!

**Comment by stevenshiau** [comment]
"my security baselines do NOT allow me to deploy an encrypted image on to multiple laptops. Because they would all have the same LUKS Master Key. " -> Have you tried to deploy them using Clonzilla, then use "cryptsetup-reencrypt" to change the master key?
https://www.golinuxcloud.com/change-luks-device-master-key-cipher-linux/#Method_2_Use_cryptsetup-reencrypt

Steven

**Comment by MarioLawrence** [comment]
> "my security baselines do NOT allow me to deploy an encrypted image on to multiple laptops. Because they would all have the same LUKS Master Key. " -> Have you tried to deploy them using Clonzilla, then use "cryptsetup-reencrypt" to change the master key? https://www.golinuxcloud.com/change-luks-device-master-key-cipher-linux/#Method_2_Use_cryptsetup-reencrypt
> 
> Steven

Thank you for the reply, Steven. 

Turns out "cryptsetup reencrypt -v <device>" (no hyphen) is a function that can work on an active partition, from within the OS. 
I got a working setup where I clone a configured VM, run reencypt to generate a new master key on the clone, and then have Clonezilla create an image from that clone. I simply keep the image ready on the smb drive for deployment to a mobile device. 

Turns out running reencrypt is faster in the VM than it is doing it on the metal. 
Reencryption took 4 minutes 30 seconds on average on the VM vs. 49 minutes on the actual hardware. 

cryptsetup reencrypt has been the faster option for me. 

Thanks again. I guess I can mark this request Solved.

**Comment by stevenshiau** [comment]
Great. It's nice to know the reencrypt function of cryptsetup works for you. Thanks for confirming that.

Steven
